aspnetcore/client-side/dotnet-interop/index.md

```diff
@@ -613,34 +613,34 @@ The preceding example displays the following output in the browser's debug conso
 Some type mappings requiring nested generic types in the [`JSMarshalAs`](xref:System.Runtime.InteropServices.JavaScript.JSMarshalAsAttribute%601) definition aren't currently supported. For example, returning a [`Promise`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) for an array such as `[return: JSMarshalAs<JSType.Promise<JSType.Array<JSType.Number>>>()]` generates a compile-time error. An appropriate workaround varies depending on the scenario, but one option is to represent the array as a <xref:System.Runtime.InteropServices.JavaScript.JSObject> reference. This may be sufficient if accessing individual elements within .NET isn't necessary and the reference can be passed to other JS methods that act on the array. Alternatively, a dedicated method can take the <xref:System.Runtime.InteropServices.JavaScript.JSObject> reference as a parameter and return the materialized array, as demonstrated by the following `UnwrapJSObjectAsIntArray` example. In this case, the JS method has no type checking, and the developer has the responsibility to ensure a <xref:System.Runtime.InteropServices.JavaScript.JSObject> wrapping the appropriate array type is passed.
 
 ```javascript
-PromisesShim.WaitGetIntArrayAsObject = function () {
+PromisesShim.waitGetIntArrayAsObject = function () {
   return new Promise((resolve, reject) => {
     setTimeout(() => {
       resolve([1, 2, 3, 4, 5] ); // Return an array from the Promise
     }, 500);
   });
 };
 
-PromisesShim.UnwrapJSObjectAsIntArray = function (jsObject) {
+PromisesShim.unwrapJSObjectAsIntArray = function (jsObject) {
   return jsObject;
 };
 ```
 
 ```csharp
 // Not supported, generates compile-time error.
-[JSImport("PromisesShim.WaitGetArray", "PromisesShim")]
+[JSImport("waitGetArray", "PromisesShim")]
 [return: JSMarshalAs<JSType.Promise<JSType.Array<JSType.Number>>>()]
 public static partial Task<int[]> WaitGetIntArray();
 
 // Workaround, take the return the call and pass it to UnwrapJSObjectAsIntArray.
 // Return a JSObject reference to a JS number array.
-[JSImport("PromisesShim.WaitGetArray", "PromisesShim")]
+[JSImport("waitGetArray", "PromisesShim")]
 [return: JSMarshalAs<JSType.Promise<JSType.Object>>()]
 public static partial Task<JSObject> WaitGetIntArrayAsObject();
 
 // Takes a JSObject reference to a JS number array, and returns the array as a C# 
 // int array.
-[JSImport("PromisesShim.WaitGetArray", "PromisesShim")]
+[JSImport("waitGetArray", "PromisesShim")]
```

Comment:
Should this be ...

```suggestion
[JSImport("unwrapJSObjectAsIntArray", "PromisesShim")]
```

Reply:
@SerratedSharp ... Could you check this one and see if my suggestion is correct or not?

Otherwise, we're just waiting on final checks of all of the code examples here. If you don't reach it, I hope to have time next week to do it.

Reply:
@SerratedSharp ... Still need to know if that update :point_up: is correct, and I have two questions. The article states ...

> If JS is loaded from a JavaScript module, then `[JSImport]` attributes must include the module name as the second parameter. For example, `[JSImport("globalThis.callAlert", "ExampleShim")]` indicates the imported method was declared in a JavaScript module named "`ExampleShim`."

... and that's the pattern that all of the examples adopt.

However, that's not what the project template does. The project template names the module ***with the file extension*** ...

```csharp
[JSImport("window.location.href", "main.js")]
```

I noticed if I change that to ...

```csharp
[JSImport("window.location.href", "main")]
```

No dice ... 💥. Should all of the module file parameter values be changed to include the file extension? For example ...

```diff
- [JSImport("incrementCounter", "PrimitivesShim")]
+ [JSImport("incrementCounter", "PrimitivesShim.js")]
```

Next question ... I hit a 💥 on my first test run with the Primatives Interop example:

> Loading module from “http://127.0.0.1:52460/PrimitivesShim.js” was blocked because of a disallowed MIME type (“”).

How do I resolve that?

Reply:
> ... Still need to know if that update ☝️ is correct

It is correct but there was a similar issue in the same file. Not sure if my comments were not appearing.  Only just realized my reviews were probably only visible to me.  Hopefully you see several suggested changes that covers this issue as well as others.

> Should all of the module file parameter values be changed to include the file extension? 

This will not work unless you also change the ImportAsync calls to include the extension in the module name.

The project template uses a file extension in JSImport because it actually names its module with *.js in the module name when calling `setModuleImports` (I personally do not use this method for declaring modules because it isn't present in other .NET WASM platforms that support JSImport):

```JS
setModuleImports('main.js', {
    window: {
        location: {
            href: () => globalThis.window.location.href
        }
    }
});
```

Reply:
When a suggestion commit is made, GH hides the comment and marks it "resolved." I merged all of your suggestions, so I think we're good here.

Reply:
Seems nicer to not use the file extension ...

```diff
- await JSHost.ImportAsync("PrimitivesShim.js", "/PrimitivesShim.js");
+ await JSHost.ImportAsync("PrimitivesShim", "/PrimitivesShim.js");
```

...

```diff
- [JSImport("incrementCounter", "PrimitivesShim.js")]
+ [JSImport("incrementCounter", "PrimitivesShim")]
```

... but *Yes!* It's working here now. 🎉

Reply:
I placed the sample app in the Blazor samples repo, which makes sense because devs are going to want to play with this code ...

https://github.com/dotnet/blazor-samples/tree/main/8.0/WASMBrowserAppImportExportInterop

I'll cross-link the code into the article next.

I hit a new 😈 ... it's in the Events Usage bits ...

> MONO_WASM: ManagedError@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:26842
Jn@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:31614
Tl<@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:182417
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[5]:0x0
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[349]:0x1fad7
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[245]:0x1bf9f
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[238]:0xf16c
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[306]:0x1e7f1
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[327]:0x1efe7
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[217]:0xcfbc
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[772]:0x44213
Bl/e[r]@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:215551
Go@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:50974
callUserCallback@http://127.0.0.1:60662/_framework/dotnet.native.8.0.7.9cskr7qo55.js:8:112495
createDotnetRuntime/safeSetTimeout/createDotnetRuntime<@http://127.0.0.1:60662/_framework/dotnet.native.8.0.7.9cskr7qo55.js:8:112626
setTimeout handler*safeSetTimeout@http://127.0.0.1:60662/_framework/dotnet.native.8.0.7.9cskr7qo55.js:8:112588
Tl<@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:155252
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[771]:0x441af
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[349]:0x1fad7
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[245]:0x1bf9f
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[238]:0xf16c
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[272]:0x1d1bd
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[3185]:0xe8756
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[2505]:0xbe4ac
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[2511]:0xbecd0
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[2535]:0xc1327
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[128]:0xa4fa
createDotnetRuntime/Module._mono_wasm_invoke_method_bound@http://127.0.0.1:60662/_framework/dotnet.native.8.0.7.9cskr7qo55.js:8:123345
kr@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:35365
Nl/e.onRuntimeInitialized/l.javaScriptExports.complete_task@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:209726
mo/<@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:42372
promise callback*mo@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:42203
Yr/<@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:39743
Tl</S<@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:178043
Tl<@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:179198
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[3]:0x0
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[349]:0x1fafb
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[245]:0x1bf9f
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[238]:0xf16c
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[272]:0x1d1bd
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[3185]:0xe8756
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[2505]:0xbe4ac
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[2511]:0xbecd0
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[2535]:0xc1327
@http://127.0.0.1:60662/_framework/dotnet.native.wasm:wasm-function[128]:0xa4fa
createDotnetRuntime/Module._mono_wasm_invoke_method_bound@http://127.0.0.1:60662/_framework/dotnet.native.8.0.7.9cskr7qo55.js:8:123345
kr@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:35365
Nl/e.onRuntimeInitialized/l.javaScriptExports.complete_task@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:209726
mo/<@http://127.0.0.1:60662/_framework/dotnet.runtime.8.0.7.s7d13jzdw4.js:3:42652

Reply:
@guardrex Not sure if source of error, but this code looks for btn1 and btn2 elements.  Probably needs a `// Assume button elements with IDs "btn1" and "btn2" exist on page`.  See this html (btn3 is not needed for the MS article):

https://github.com/SerratedSharp/CSharpWasmRecipes/blob/serrated/jsimport-msarticle/WasmBrowser.Recipes.WasmClient/wwwroot/index.html